php/class-wp-seo-social-wp-seo-filters.php

```diff
@@ -0,0 +1,176 @@
+			'og_image'         => '',
+			'og_type'          => '',
+		);
+		$array = array_merge( $array, $extra_fields );
+		return $array;
+	}
+
+	/**
+	 * Filter the whitelisted fields so ours validate.
+	 *
+	 * @param array $array Array of whitelisted fields.
+	 * @return Array of filtered whitelisted fields.
+	 */
+	public function filter_wp_seo_whitelisted_fields( $array ) {
+		$array = array_merge( $array, WP_SEO_Social_Settings()->fields_to_whitelist );
+		return $array;
+	}
+
+	/**
+	 * Filter the WP SEO metabox heading.
+	 */
+	public function filter_wp_seo_box_heading() {
+		return __( 'Search & Social Optimization', 'wp-seo-social' );
+	}
+
+	/**
+	 * Filter arbitrary tags output to print out OG tags.
+	 *
+	 * @param array $arbitrary_tags Array of any existing arbitrary tags.
+	 * @return Filtered array of tags plus arbitrary tags.
+	 */
+	public function filter_wp_seo_arbitrary_tags( $arbitrary_tags ) {
+		$pretags = array();
+		$tags = array();
+		$key = WP_SEO()->get_key();
+		if ( is_singular() ) {
+			if ( WP_SEO_Settings()->has_post_fields( $post_type = get_post_type() ) ) {
+				foreach ( WP_SEO_Social_Settings()->fields_to_whitelist as $field ) {
+					$field_string = get_post_meta( get_the_ID(), '_meta_' . $field, true );
+					$pretags[ $field ] = $field_string;
+				}
+			}
+		} elseif ( is_category() || is_tag() || is_tax() ) {
+			if ( WP_SEO_Settings()->has_term_fields( $taxonomy = get_queried_object()->taxonomy )
+				&& $option = get_option( WP_SEO()->get_term_option_name( get_queried_object() ) )
+			) {
+				foreach ( WP_SEO_Social_Settings()->fields_to_whitelist as $field ) {
+					if ( isset( $option[ $field ] ) ) {
+						$field_string = $option[ $field ];
+						$pretags[ $field ] = $field_string;
+					}
+				}
+			}
+		}
+		foreach ( WP_SEO_Social_Settings()->fields_to_whitelist as $field ) {
+			if ( empty( $pretags[ $field ] ) ) {
+				/**
+				 * Filter the format strings of whitelisted custom tags.
+				 *
+				 * @param  string 		The format string retrieved from the settings.
+				 * @param  string $key	The key of the setting retrieved.
+				 */
+				$field_string = apply_filters( 'wp_seo_meta_' . $field . '_format', WP_SEO_Settings()->get_option( $key . '_' . $field ), $key );
+				$meta_field_value = WP_SEO()->format( $field_string );
+				$pretags[ $field ] = $meta_field_value;
+			} else {
+				$pretags[ $field ] = WP_SEO()->format( $pretags[ $field ] );
+			}
+		}
+		foreach ( $pretags as $key => $value ) {
+			if ( $value && ! is_wp_error( $value ) ) {
+				if ( 'og_image' === $key ) {
+					$og_img_src = wp_get_attachment_image_src( $value, 'og_image' );
+					if ( ! empty( $og_img_src[0] ) ) {
+						$tags[] = array(
+							'name' => $key,
+							'content' => $og_img_src[0],
+						);
+					}
+				} else {
+					$tags[] = array(
+						'name' => $key,
+						'content' => $value,
+					);
+				}
+			}
+		}
+		$arbitrary_tags = array_merge(
+			$arbitrary_tags,
+			$tags
+		);
+		return $arbitrary_tags;
```

Comment:
same as above, you could just return the array_merge itself if you are not going to validate it.